test/TypeCheckerTests.hs

```diff
@@ -205,6 +206,15 @@ testCompleteBoardEq = TestCase (
   allPassTC [testGame [(BVal (Sig "b" (Plain boardxt)) [PosDef "b" (ForAll ("x")) (ForAll ("y")) (I 1)] dummyPos)]]
   )
 
+testFeqMismatch :: Test
+testFeqMismatch = TestCase (
+  (assertBool "Verifies that a function equation is not allowed for a value equation signature") $
+  let eqn = (Val (Sig "b" (Plain boardxt)) (Feq "b" (Pars ["x", "y"]) (I 1)) dummyPos) in
+  case tc $ testGame [eqn] of
+    (Tc False _ [(_, Error (TE (SigBadFeq _ _ _)) _ _)] _) -> True
+    _                                      -> False
+  )
+
```

Comment:
Didn't realize we had function equations associating with value equation signatures somehow...

Reply:
Unless I am misunderstanding you, we don't! This test checks that that produces an error.